packages/swap/src/event-handlers/egressAmountZero.ts

```diff
@@ -0,0 +1,39 @@
+import { z } from 'zod';
+import { assetChains } from '@/shared/enums';
+import { u64 } from '@/shared/parsers';
+import type { EventHandlerArgs } from '.';
+
+const egressAmountZeroArgs = z.object({
+  swapId: u64,
+});
+
+export type EgressAmountZeroEvent = z.input<typeof egressAmountZeroArgs>;
+
+export default async function egressAmountZero({
+  prisma,
+  event,
+}: EventHandlerArgs): Promise<void> {
+  const { swapId } = egressAmountZeroArgs.parse(event.args);
+
+  const swap = await prisma.swap.findUniqueOrThrow({
+    where: { nativeId: swapId },
+    include: { swapDepositChannel: true },
+  });
+  const channel = swap.swapDepositChannel;
+
+  if (!channel) {
```

Comment:
i think this error can happen for smart-contract swaps too 😕 so we cannot throw an error here

Reply:
maybe we can adapt the `FailedSwap` entity to be connected to a channel or a swap? not sure if thats the best solution but it would be nice to expose the error from the api for smart contract swaps too

Reply:
yeah good point, i missed that. its already connected to a swap channel. i think we'd have to connect a `FailedSwap` also to a `Swap` to account for those scenarios.